### PR TITLE
Add note when updating system.memory

### PR DESCRIPTION
**Pull request by BenVillalobos**
### Context
We have customer projects break when we update system.memory, we should make it easier for customers to figure out these breakages when they're due to us updating system.memory.

### Changes Made
Add a simple note on our PackageReference to System.Memory, with guidelines.

### Notes
PR is WIP.

### CODE REVIEW

**Comment by BenVillalobos** [comment]
Note to self: Also add the note to the relevant places here: https://github.com/dotnet/msbuild/pull/7680/files

File: eng/Packages.props [hunk 1]
```diff
@@ -18,8 +18,14 @@
     <PackageReference Update="NuGet.Frameworks" Version="$(NuGetBuildTasksVersion)" />
     <PackageReference Update="System.Collections.Immutable" Version="$(SystemCollectionsImmutableVersion)" />
     <PackageReference Update="System.Configuration.ConfigurationManager" Version="$(SystemConfigurationConfigurationManagerVersion)" />
-    <PackageReference Update="System.Net.Http" Version="$(SystemNetHttpVersion)" />
+
+    <!-- 
+        Modifying the version of System.Memory is very high impact and often causes downstream breaks on repos that depend on us.
```
Comment (Forgind) [inline eng/Packages.props:23]: Should this instead be where we define SystemMemoryVersion?

File: eng/Versions.props [hunk 2]
```diff
@@ -27,6 +27,10 @@
   <PropertyGroup>
     <SystemCollectionsImmutableVersion>6.0.0</SystemCollectionsImmutableVersion>
     <SystemConfigurationConfigurationManagerVersion>6.0.0</SystemConfigurationConfigurationManagerVersion>
+    <!-- 
+        Modifying the version of System.Memory is very high impact and often causes downstream breaks on repos that depend on us.
+        When updating the version of System.Memory, file a breaking change here: https://github.com/dotnet/docs/issues/new?assignees=gewarren&labels=breaking-change%2CPri1%2Cdoc-idea&template=breaking-change.yml&title=%5BBreaking+change%5D%3A+
```
Comment (Forgind) [inline eng/Versions.props:32]: I believe rainersigwald suggested doing some do diligence to figure out why it was such a breaking change in this case. Have you done that?

File: eng/Versions.props [hunk 3]
```diff
@@ -27,6 +27,10 @@
   <PropertyGroup>
     <SystemCollectionsImmutableVersion>6.0.0</SystemCollectionsImmutableVersion>
     <SystemConfigurationConfigurationManagerVersion>6.0.0</SystemConfigurationConfigurationManagerVersion>
+    <!-- 
+        Modifying the version of System.Memory is very high impact and often causes downstream breaks on repos that depend on us.
```
Comment (rainersigwald) [inline eng/Versions.props:31]:
```suggestion
        Modifying the version of System.Memory is very high impact and causes downstream breaks in third-party tooling that uses the MSBuild API.
```

File: eng/Versions.props (hunk 2 above)
Comment (rainersigwald) [inline eng/Versions.props:32]: Is this sufficient? Should we also post a VS breaking change? Should we say "try not to do this but"?

**Review by rainersigwald — COMMENTED** [review]
After figuring out what was going on in https://github.com/dotnet/msbuild/issues/7873#issuecomment-1227332842, I think this is a good path forward, with some wording tweaks.

**Comment by rainersigwald** [comment]
I think we should go forward with this path--maybe with even stronger warnings since it's subtle and painful. @BenVillalobos can you adopt some of the changes and we'll get it merged?

File: eng/Versions.props (hunk 2 above)
Comment (BenVillalobos) [reply eng/Versions.props:32]: Not a bad idea. I searched around a bit, but couldn't find a link to file VS breaking changes. Do you have one handy?

Comment (rainersigwald) [reply eng/Versions.props:32]: I think https://dev.azure.com/devdiv/DevDiv/_wiki/wikis/DevDiv.wiki/1796/How-to-add-a-Known-Issue (internal to Microsoft) is the best we have

**Comment by BenVillalobos** [comment]
Added link to internal guidelines, should be good to go

**Review by rainersigwald — APPROVED** [review]

**Pull request merged**